src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -41,7 +41,8 @@
     private final boolean skipNegotiate;
     private Single<String> accessTokenProvider;
     private Single<String> redirectAccessTokenProvider;
-    private final Map<String, String> headers = new HashMap<>();
+    private Map<String, String> headers;
```

Comment:
Since this isn't assigned outside of the ctor, can this still be marked final?

Reply:
It can't because it's conditionally set

Reply:
You could just set it unconditionally though, right?

Reply:
Sure